spec/src/main/asciidoc/app-programming-model.adoc

```diff
@@ -61,8 +61,10 @@ in <<architecture#rest-api>>.
 * The implementation must detect if multiple annotations declare the same gauge (with the same `MetricID`)
 and throw an `IllegalArgumentException` if such duplicate exists
 ** See <<architecture#reusing_metrics,reusing of metrics>> for more details.
-* The implementation must flag and reject metrics upon registration if the metadata information being registered is not equivalent to the metadata information that has already been registered under the given metric name (if it already exists).
-** All metrics of a given metric name must be associated with the same metadata information
+* The implementation must reject metrics upon registration if the metadata information being registered is not equivalent to the metadata information that has already been registered under the given metric name (if it already exists).
+** All metrics of a given metric name must be associated with the same metadata information.
+** The implementation must throw an `IllegalArgumentException` when the metric is rejected.
+* The implementation must reject metrics upon registration if the names of the tags are not equivalent to those already registered under the given metric name (if it already exists). The values of the tags may differ, but the name/key of the tags must be the same.
```

Comment:
suggest...

The implementation must reject metrics upon registration if the set of tag names specified is not the same as the set of tag names used in prior registrations of metrics with the same metric name.